hs/src/Reach/Optimize.hs

```diff
@@ -393,13 +394,15 @@ instance Optimize DLExpr where
     DLE_Claim at fs t a m -> do
       a' <- opt a
       let meh = return $ DLE_Claim at fs t a' m
-      case (t, a') of
-        (CT_Possible, _) -> meh
-        (_, DLA_Literal (DLL_Bool True)) -> do
+      isSim <- asks eSimulate
```

Comment:
You're going to have to consider some operations to be impure under simulation, but I believe that is going to be part of ac, not opt, so that's pretty groddy.